### PR TITLE
Turn the validate.go into a testunit which is a class

### PR DESCRIPTION
**Pull request by liangchenye**
And move the convert into rkt, since each runtime should know what it need to do before running a 'standard oci bundle'

### CODE REVIEW

File: testunit.go [hunk 1]
```diff
@@ -10,102 +9,88 @@ import (
...
+func TestUnitNew(name string, args string, desc string) (unit TestUnit) {
+	unit.Name = name
+	unit.Args = args
+	unit.Description = desc
 
-	logrus.Debugf("Args to the ocitools generate: ")
-	for _, a := range args {
-		logrus.Debugln(a)
-	}
-	Mutex.Lock()
```
Comment (zenlinTechnofreak) [inline testunit.go]: @liangchenye  Why u delete the mutex lock here?

Comment (liangchenye) [inline testunit.go]: It was not deleted, seems 'git diff' is not readable.
You can read the original code:
https://github.com/liangchenye/oct/blob/master/testunit.go

Comment (zenlinTechnofreak) [inline testunit.go]: OK， LGTM.

File: testunit.go [hunk 2]
```diff
@@ -10,102 +9,88 @@ import (
...
+	unit.PrepareBundle()
+	_, err := unit.Runtime.StartRT(unit.BundleDir)
+	return err
+	/* TODO: what is testopt used for ?
+	testopt := utils.GetAfterNStr(configArgs, "--args=./runtimetest --args=", 3)
+	switch testopt {
+	case "vna":
+		err = hooks.SetPostStartHooks(out, hooks.NamespacePostStart)
+	default:
 	}
```
Comment (wangkirin) [inline testunit.go]: @liangchenye `testopt` is the `specs` item to be test , which is the argument after `./runtimetest`, for some item we should add hooks before or after  we start container runtime

File: testunit.go [hunk 3]
```diff
@@ -154,6 +139,49 @@ func prepareBundle(validateObj string) {
...
+			continue
+		} else {
+			resArray[i-1] = "--" + strings.TrimSpace(arg)
+		}
+	}
+	return resArray
+}
+
 func copy(dst string, src string) error {
 	in, err := os.Open(src)
```
Comment (wangkirin) [inline testunit.go]: @zenlinTechnofreak May be we can consider put  `copy` function into `utils/file.go`

**Comment by liangchenye** [comment]
@zenlinTechnofreak @wangkirin  conflict resolve.

**Pull request merged**